tests/reassociate_mul.bril

```diff
@@ -0,0 +1,15 @@
+# (x * (5 * (y * (6 * z)))) -> (x * (y * (z * 30)))
+@mul_reassoc (x: int, y: int, z: int): int {
```

Comment:
might be a problem if main doesn't use this function- might get dead code eliminated in the future

Reply:
We might similarly have an issue if we do call the function and an inliner determines it to be inlineable :)

I lean towards not doing deadcode on functions for now, unless we're pushing for codegen size as a metric of success which I don't think we should/are.

Reply:
Good point, lets just deal with this if it becomes a problem